runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1197,13 +1239,36 @@ public State getState() {
     }
 
     public void setState(State state) {
+      Instant now = clock.get();
```

Comment:
I believe this function is called multiple times per work item. We need to performance test this to make sure pipelines don't get more expensive (and we don't make e2e latency even worse)

Reply:
Ack. Will take a while and post results here.

Reply:
I've run two different pipelines that include this PR with Google Cloud Profiler enabled. They spent 105µs out of 18.65s overall and 4.45ms out of 16.43s overall modifying/reading Work.State, respectively. This is a negligible fraction of CPU time. If there's any at all, any performance regression would be negligible, too.